### PR TITLE
o/aspectstate: get local aspect-bundle assertions

### PR DESCRIPTION
**Pull request by MiguelPires**
* 0e59baa17d606d5e6e5c25249faccc4f22b50335 changes aspectstate to get a local aspect-bundle assertion instead of using the mock. It also moves the transaction creation into aspectstate since it now needs to use the real schema from the assertion
* 225302e0ca7b1d266e39fd0f9dbdf91a94d81a5c adds a signed aspect-bundle assertion and uses it to replace the mock
* 1a914e31a1c64c4dcbea5df3432c5e0aac0cff78 removes the mocked assertion previously used in unit and spread tests

### CODE REVIEW

File: aspects/aspects.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func notFoundErrorFrom(a *Aspect, op, request, errMsg string) *NotFoundError {
 		BundleName: a.bundle.Name,
 		Aspect:     a.Name,
 		Operation:  op,
-		Request:    request,
+		Request:    fmt.Sprintf("%q", request),
```
Comment (pedronis) [inline aspects/aspects.go:86]: why this change? it's a bit odd tbh

File: overlord/aspectstate/aspectstate.go [hunk 2]
```diff
@@ -22,83 +22,125 @@ import (
...
+			keys = append(keys, k)
+		}
+
 		return &aspects.NotFoundError{
 			Account:    account,
 			BundleName: bundleName,
 			Aspect:     aspect,
 			Operation:  "set",
-			Request:    field,
+			Request:    strutil.Quoted(keys),
```
Comment (pedronis) [inline overlord/aspectstate/aspectstate.go:51]: ah, it would make more sense perhaps to make Request into Requests []string ?

File: overlord/aspectstate/aspectstate.go [hunk 3]
```diff
@@ -137,3 +182,18 @@ func updateDatabags(st *state.State, account, bundleName string, databag aspects
 	st.Set("aspect-databags", databags)
 	return nil
 }
+
+func getBundle(st *state.State, account, bundleName string) (*aspects.Bundle, error) {
```
Comment (pedronis) [inline overlord/aspectstate/aspectstate.go:186]: usually this kind of helper lives in assertstate

**Review by pedronis — COMMENTED** [review]
thanks, did a first pass

File: aspects/aspects.go (hunk 1 above)
Comment (MiguelPires) [reply aspects/aspects.go:86]: It's because aspectstate now also uses this error with potentially more than one request and quoting that looked strange. But you're right, in hindsight this is very hacky. I'll change the field to acomodate it

File: overlord/aspectstate/aspectstate.go (hunk 2 above)
Comment (MiguelPires) [reply overlord/aspectstate/aspectstate.go:51]: ah sorry, I didn't see this comment before replying to the other one. Yes, I'll change the field to reflect the new usage as well

File: asserts/aspect_bundle.go [hunk 4]
```diff
@@ -106,3 +106,18 @@ func assembleAspectBundle(assert assertionBase) (Assertion, error) {
 		timestamp:     timestamp,
 	}, nil
 }
+
+// GetLocalAspectBundle returns a locally stored aspect bundle identified by
+// the account and bundle name.
+func GetLocalAspectBundle(db RODatabase, account, bundleName string) (*aspects.Bundle, error) {
```
Comment (pedronis) [inline asserts/aspect_bundle.go:112]: this should just be AspectBundle. For the doc comment follow the pattern of something like SnapDeclaration

**Review by pedronis — COMMENTED** [review]
thanks for the changes, remark

File: asserts/aspect_bundle.go (hunk 4 above)
Comment (pedronis) [reply asserts/aspect_bundle.go:112]: sorry, this is the wrong place =>  I wrote I think assertstate

**Review by pedronis — APPROVED** [review]
thank you

File: overlord/aspectstate/aspectstate.go [hunk 5]
```diff
@@ -22,83 +22,125 @@ import (
...
+	}
+
+	results := make(map[string]interface{}, len(fields))
+	for _, field := range fields {
+		value, err := asp.Get(tx, field)
+		if err != nil {
+			if errors.Is(err, &aspects.NotFoundError{}) && len(fields) > 1 {
+				// keep looking; return partial result if only some fields are found
+				continue
+			} else {
```
Comment (andrewphelpsj) [inline overlord/aspectstate/aspectstate.go:103]: This else could be removed

**Review by andrewphelpsj — APPROVED** [review]
Thanks, looks good to me!
